recipes/mocknetworkaccessmanager/all/conanfile.py

```diff
@@ -0,0 +1,76 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.52.0"
+
+
+class MockNetworkAccessManagerConan(ConanFile):
+    name = "mocknetworkaccessmanager"
+    description = "Mocking network communication for Qt applications"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.com/julrich/MockNetworkAccessManager"
+    topics = ("qt", "mock", "network", "QNetworkAccessManager", "unit test", "test", "header-only")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("qt/5.15.11", transitive_headers=True)
```

Comment:
Could you please clarify two things:
- What header in MockNetworkAccessManager needs a transitive header from Qt
- Why not use Qt6 instead of Qt5. I see the preference is Qt6 in the project: https://gitlab.com/julrich/MockNetworkAccessManager/-/blob/0.12.0/CMakeLists.txt#L18

Reply:
If I don't set Qt transitive header, I have an error including [QtGlobal](https://gitlab.com/julrich/MockNetworkAccessManager/-/blob/0.12.0/MockNetworkAccessManager.hpp#L32)

Regarding Qt6, I can't manage to compile because Qt complains that it requires a C++17 compiler.

It is the case according to my profile:

```
$ conan create "mocknetworkaccessmanager/all" --version 0.12.0 \
		--user=cci_testing \
		--channel=testing \
		-pr:b Darwin.txt \
		-pr:h Darwin.txt \
		--build mocknetworkaccessmanager/0.12.0@cci_testing/testing \
		--update \


======== Exporting recipe to the cache ========
mocknetworkaccessmanager/0.12.0@cci_testing/testing: Exporting package recipe: /Users/martindelille/dev/conan/cci/recipes/mocknetworkaccessmanager/all/conanfile.py
mocknetworkaccessmanager/0.12.0@cci_testing/testing: exports: File 'conandata.yml' found. Exporting it...
mocknetworkaccessmanager/0.12.0@cci_testing/testing: Calling export_sources()
mocknetworkaccessmanager/0.12.0@cci_testing/testing: Copied 1 '.py' file: conanfile.py
mocknetworkaccessmanager/0.12.0@cci_testing/testing: Copied 1 '.yml' file: conandata.yml
mocknetworkaccessmanager/0.12.0@cci_testing/testing: Copied 2 '.patch' files: 0.12.0-0002-fix-target-without-location.patch, 0.12.0-0001-add-install-step.patch
mocknetworkaccessmanager/0.12.0@cci_testing/testing: Exported to cache folder: /Users/martindelille/.conan2/p/mockn93efa29b786ce/e
mocknetworkaccessmanager/0.12.0@cci_testing/testing: Exported: mocknetworkaccessmanager/0.12.0@cci_testing/testing#ef939dff2372ad0714d72577a0f70137 (2024-02-15 09:06:17 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=14.0
os=Macos

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=14.0
os=Macos
...
```

Reply:
@MartinDelille C++17 is a blocker for you? Otherwise, I would suggest using it by default.

Or, you just add an option to install one or another, and pass the definition `FORCE_QT5` to cmake

Reply:
No I'm personnally using Qt 6 and C++17 I just don't understand why my `conan create ...` command fails:

```

In file included from /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/QtCore:4:
In file included from /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qglobal.h:26:
In file included from /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qtcoreexports.h:7:
/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qcompilerdetection.h:1239:6: error: "Qt requires a C++17 compiler"
#    error "Qt requires a C++17 compiler"
     ^
In file included from /Users/martindelille/.coIn file included from /Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/tests/MyNetworkClientTest_autogen/mocs_compilation.cpp:2nan2/p/b/mocknb4f6377b32d6e/b/src/tests/CompatibilityTest.cpp:
In file included from :1In file included from :
/Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/src/DummyMain.cpp:1:
In file included from /Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/src/MockNetworkAccessManager.hpp:32In file included from :
/Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/tests/MyNetworkClientTest_autogen/YCDLW3T4OG/moc_MockNetworkAccessManager.cppIn file included from :/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/QtGlobal:1:
In file included from In file included from /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qglobal.h/Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/src/MockNetworkAccessManager.hpp:32/Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/MockNetworkAccessManager_autogen/mocs_compilation.cpp:
:9:26:
:
In file included from In file included from /Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/tests/MyNetworkClientTest_autogen/YCDLW3T4OG/../../../../../src/MockNetworkAccessManager.hpp:/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qtcoreexports.h32::
7In file included from /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/QtGlobal:1In file included from :
2In file included from :
/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qglobal.h:
:/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/QtGlobal26In file included from /Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/MockNetworkAccessManager_autogen/EWIEGA46WW/moc_MockNetworkAccessManager.cpp::19:
:
In file included from In file included from /Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/MockNetworkAccessManager_autogen/EWIEGA46WW/../../../../src/MockNetworkAccessManager.hpp/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qcompilerdetection.h:
:/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qglobal.h:26:
1239:In file included from :6In file included from :/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qtcoreexports.h :7:
/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qtcoreexports.h/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qcompilerdetection.h32:1239::7:
In file included from error: /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/QtGlobal"Qt requires a C++17 compiler"6: :
:
1#    error "Qt requires a C++17 compiler"/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qcompilerdetection.h
::
error: In file included from 1239/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qglobal.h     ^"Qt requires a C++17 compiler"

::626#    error "Qt requires a C++17 compiler":
:
In file included from  /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qtcoreexports.h:7:
     ^
/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qcompilerdetection.h:1239:6: error: "Qt requires a C++17 compiler"
#    error "Qt requires a C++17 compiler"
     ^
In file included from /Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/tests/CompatibilityTest_autogen/mocs_compilation.cpp:error: 2:
In file included from "Qt requires a C++17 compiler"/Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/tests/CompatibilityTest_autogen/YCDLW3T4OG/moc_MockNetworkAccessManager.cpp:9:

In file included from /Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/build/Release/tests/CompatibilityTest_autogen/YCDLW3T4OG/../../../../../src/MockNetworkAccessManager.hpp:32:
In file included from /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/QtGlobal:1:
In file included from #    error "Qt requires a C++17 compiler"/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qglobal.h
:     ^
26:
In file included from /Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qtcoreexports.h:7:
/Users/martindelille/.conan2/p/b/qte03224e3525ca/p/include/QtCore/qcompilerdetection.h:1239:6: error: "Qt requires a C++17 compiler"
#    error "Qt requires a C++17 compiler"
     ^
In file included from /Users/martindelille/.conan2/p/b/mocknb4f6377b32d6e/b/src/tests/MyNetworkClientTest.cpp:3:
```